### PR TITLE
feat(sdk-python): Port in-line expressions to Python SDK

### PR DESCRIPTION
**Pull request by Snarr**
This PR ports the in-line expression changes to the Python SDK.

### CODE REVIEW

File: sdk-python/littlehorse/workflow.py
Comment (sauljabin) [inline sdk-python/littlehorse/workflow.py:510]: 1 underscore for internal 2 for language related ones.

example `__init__` 2 underscore

Comment (sauljabin) [inline sdk-python/littlehorse/workflow.py:510]: 1 underscore for internal 2 for language related ones.

example __init__ 2 underscore

File: sdk-python/littlehorse/workflow.py
Comment (sauljabin) [inline sdk-python/littlehorse/workflow.py:354]: if the default value is None use `= None` here

Optional[WorkflowThread]

**Review by sauljabin — APPROVED** [review]
I have approve it, but before merging check if it is possible to get rid of the dependency at WfRunVariable and use the WorkflowCondition.

The other WfRunVariable parameter are attributes of the class, WorkflowThread would be a dependency that maybe we can avoid, if we can the better

File: sdk-python/littlehorse/workflow.py
Comment (Snarr) [reply sdk-python/littlehorse/workflow.py:354]: I agree with your proposal, it is unnecessary to depend on the parent condition methods when creating a new WorkflowCondition is so simple. If the code for creating a new WorkflowCondition were more complex, it may make sense to centralize it in one place and rely on this dependency injection or another class to hold the methods.

There is *one* additional method that may not make sense to port without injecting the parent thread dependency:

```python
    @Override
    public void assignTo(Serializable rhs) {
        parent.mutate(this, VariableMutationType.ASSIGN, rhs);
    }
```

Without injecting the parent WorkflowThread at the constructor, this Thread may not make sense or be easy to use. Instead, we would have to pass it to the method as an argument, which would not save the user much code and (in my opinion) be less readable.

Comment (sauljabin) [reply sdk-python/littlehorse/workflow.py:354]: You are right, I did not see the mutate

File: sdk-python/littlehorse/workflow.py
Comment (Snarr) [reply sdk-python/littlehorse/workflow.py:144]: Forgot to remove this, good catch

**Pull request merged**
